src/test/mir-opt/const-promotion-extern-static.rs

```diff
@@ -14,7 +14,7 @@ fn main() {}
 // START rustc.FOO.PromoteTemps.before.mir
 // bb0: {
 // ...
-//     _5 = const Scalar(alloc1+0) : &i32;
+//     _5 = const {alloc1+0: &i32};
```

Comment:
If `+0` is common, maybe it should be omitted?
And maybe it should say `&alloc1: &i32`, and for a raw pointer be `&raw const alloc1: *const i32`.
Or we could have `&(alloc1: i32)` (and no braces?), not sure which one is nicer.

Reply:
> If +0 is common, maybe it should be omitted?

I feel that could be confusing... it is more consistent to always have the offset.

Reply:
I *almost* feel like `{alloc1+12: &i32}` should be this monster:
```rust
&*((&raw const alloc1).add(12) as *const i32)
```
Is that too much? I don't know.

I care much more about showing what `alloc1` actually consists of, somewhere in the same file, than the syntax here though.

Reply:
OMG WTF.^^
At least for Miri tracing, that would be a disaster -- way too verbose. And it's still not actual Rust code so why bother?

Reply:
> And it's still not actual Rust code so why bother?

I think I agree for miri purposes, but I'd prefer if constants in MIR were as close to Rust as possible.

Reply:
I'm going to address this when also starting to emit the allocations